packages/platforms/delta/arm64/tx4810-dn/modules/builds/src/arm64-delta-tx4810-dn-led.c

```diff
@@ -0,0 +1,391 @@
+/*
+ * A LED driver for the delta_tx4810_dn_led
+ *
+ * Copyright (C) 2021 Delta Networks, Inc.
+ *
+ * Chenglin Tsai <chenglin.tsai@deltaww.com>
+ *
+ * Based on:
+ * accton_as4610_leds.c
+ * Copyright (C) 2016 Accton Technology Corporation.
+ * Brandon Chuang <brandon_chuang@accton.com.tw>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.     See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+ */
+
+#include <linux/module.h>
+#include <linux/kernel.h>
+#include <linux/init.h>
+#include <linux/platform_device.h>
+#include <linux/err.h>
+#include <linux/leds.h>
+#include <linux/slab.h>
+#include <linux/gpio.h>
+#include <linux/bitops.h>
+
+
+#define DRVNAME "tx4810_dn_led"
+
+/* LED related data
+ */
+enum led_type {
+    LED_TYPE_PSU,
+    LED_TYPE_FAN,
+    LED_TYPE_SYS,
+    NUM_OF_LED
+};
```

Comment:
Although a Linux kernel module, the whole file does not follow the [Linux kernel coding style](https://www.kernel.org/doc/html/latest/process/coding-style.html).